### ISSUE TITLE
Account (i.e. party) not blocked when proof of work is reused in same block

### ISSUE THREAD

**Issue by MuthuVega** [vegaprotocol/vega#6384]
Spec Ref: https://github.com/vegaprotocol/specs/blob/master/protocol/0072-SPPW-spam-protection-PoW.md

**Acceptance Criteria:**

Reusing the same PoW for several messages is detected and leads to a blocking of the account ([0072-SPPW-002](https://github.com/vegaprotocol/specs/blob/master/protocol/0072-SPPW-spam-protection-PoW.md#0072-SPPW-002))

**Scenario Setup:**

spam.pow.numberOfTxPerBlock = 2
Submit 100 vote on proposal transactions with the same proof of work details where by they are all added in the same block

**Observed Behaviour:**

A significant number of transactions are added to the block and the party is NOT blocked for 4 epochs.

As it can be seen from the link - a large number of transactions have made it to the block - **673ED843B445BA5FABB0F9D6CE609A97656E8ECCC8218D746376ED5A402E3D4D**

https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/12612/artifact/test_logs/workspace.tests.wallet.vegawallet_test/test_submitting_multiple_transactions_reusing_proof_of_work_are_rejected/BUS_EVENT_TYPE_TX_ERROR.txt

**Expected Behaviour:**

Transactions are rejected as span when the number of transactions in the block is greater than spam.pow.numberOfTxPerBlock 
Also, the party should be banned from submitting any transaction for 4 epochs.

Jenkins Run:
Test Marker : bug_reuse_pow
System Test Branch: feat_0072_spam_tests_2
https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/12612/

**Comment by ze97286** [comment]
well, sure they made it to the block because *before* the block they have not been seen. Only after this block ends the single instance of this tx will make it into "seen" and be able to pre-reject future transactions from going into blocks. You can see that in fact only 2 transactions were accepted and the rest were rejected which is fine. 

with regard to banning, I see it in the AC, but can you point to where in the spec this is required?

**Comment by MuthuVega** [comment]
Thanks @ze97286 

Not too sure where you are seeing only 2 accepted and rest rejected.

I can see approx 50 transactions with the same block ID **673ED843B445BA5FABB0F9D6CE609A97656E8ECCC8218D746376ED5A402E3D4D**- in this link 
https://jenkins.ops.vega.xyz/job/common/job/system-tests-wrapper/12612/artifact/test_logs/workspace.tests.wallet.vegawallet_test/test_submitting_multiple_transactions_reusing_proof_of_work_are_rejected/BUS_EVENT_TYPE_TX_ERROR.txt

And I assumed that if there was a block ID , these transactions have been added to the block.
Also, whats the functionality of the network param **spam.pow.numberOfTxPerBlock** for me to test properly.


> with regard to banning, I see it in the AC, but can you point to where in the spec this is required

Can't see a reference to this in the spec - BUT can only see it in the AC - please let us know if you think this is incorrect and I shall raise it with Protocol Design

**Comment by ze97286** [comment]
yes, please ask about the ban in protocol design

**Comment by MuthuVega** [comment]
Happy to close this as there is another ticket opened for this - https://github.com/vegaprotocol/specs/issues/1291
